src/bd/inlining/InlinableNodes.java

```diff
@@ -123,7 +123,7 @@ private void initializeFactories(
    *           triggered by the inlining logic.
    */
   public <N extends Node, S extends ScopeBuilder<S>> N inline(final Id selector,
-      final List<N> argNodes, final S builder, final long coord)
+      final N[] argNodes, final S builder, final long coord)
```

Comment:
What's the benefit of using an array instead of a list? The fact that it was unnecessary in the first place? Is there a specific relation to the string supernode I don't see?

Reply:
This is already merged on master. I just stumbled over this when working on it here. In most cases.
We need an array later on. So, allocating the list of binary nodes was extra work, which is now avoided.
But yes, this is unrelated to this PR, and will be gone after rebasing.